### PR TITLE
Use AssertJ instead of JUnit assertions.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java [hunk 1]
```diff
@@ -542,36 +545,37 @@ private void assertNotANumber(String s) throws IOException {
...
   }
 
   @Test public void peekLongMinValue() throws IOException {
     JsonReader reader = newReader("[-9223372036854775808]");
     reader.setLenient(true);
     reader.beginArray();
-    assertEquals(NUMBER, reader.peek());
-    assertEquals(-9223372036854775808L, reader.nextLong());
+    assertThat(reader.peek()).isEqualTo(NUMBER);
+    Object expected = -9223372036854775808L;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: What happened here?

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java [hunk 2]
```diff
@@ -415,29 +415,32 @@
...
+    assertThat(reader.nextInt()).isEqualTo(0);
+    assertThat(reader.nextDouble()).isEqualTo(0.0, Offset.offset(0d));
+    assertThat(reader.nextLong()).isEqualTo(1L);
+    assertThat(reader.nextInt()).isEqualTo(1);
+    assertThat(reader.nextDouble()).isEqualTo(1.0, Offset.offset(0d));
+    Object expected2 = -1L;
+    assertThat(reader.nextLong()).isEqualTo(expected2);
+    Object expected1 = -1;
+    assertThat(reader.nextInt()).isEqualTo(expected1);
+    double expected3 = -1.0;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: And these

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java [hunk 3]
```diff
@@ -338,17 +336,19 @@
...
-    assertEquals(2.2250738585072014E-308, reader.nextDouble(), 0d);
-    assertEquals(3.141592653589793, reader.nextDouble(), 0d);
-    assertEquals(2.718281828459045, reader.nextDouble(), 0d);
+    double expected1 = -0.0;
+    assertThat(reader.nextDouble()).isEqualTo(expected1, Offset.offset(0d));
+    assertThat(reader.nextDouble()).isEqualTo(1.0, Offset.offset(0d));
+    assertThat(reader.nextDouble()).isEqualTo(1.7976931348623157E308, Offset.offset(0d));
+    assertThat(reader.nextDouble()).isEqualTo(4.9E-324, Offset.offset(0d));
+    assertThat(reader.nextDouble()).isEqualTo(0.0, Offset.offset(0d));
+    double expected = -0.5;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: More

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java [hunk 4]
```diff
@@ -75,32 +71,32 @@
...
-    assertNull(map.put("f", "froyo"));
-    assertEquals(3, map.size());
-
-    assertEquals("donut", map.get("d"));
-    assertEquals("donut", map.put("d", "done"));
-    assertEquals(3, map.size());
+    assertThat(map.put("d", "donut")).isNull();
+    assertThat(map.put("e", "eclair")).isNull();
+    assertThat(map.put("f", "froyo")).isNull();
+    assertThat(map.size()).isEqualTo(3);
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: `assertThat(map).hasSize(3)`

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java [hunk 5]
```diff
@@ -291,6 +287,6 @@ private String toString(Node<?, ?> root) {
     for (T t : actual) {
       actualList.add(t);
     }
-    assertEquals(Arrays.asList(expected), actualList);
+    assertThat(actualList).isEqualTo(Arrays.asList(expected));
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: `assertThat(actualList).containsExactly(expected)`

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java [hunk 6]
```diff
@@ -390,9 +390,9 @@
     JsonReader reader = newReader(json);
     reader.setLenient(true);
     reader.beginArray();
-    assertTrue(Double.isNaN(reader.nextDouble()));
-    assertEquals(Double.NEGATIVE_INFINITY, reader.nextDouble(), 0d);
-    assertEquals(Double.POSITIVE_INFINITY, reader.nextDouble(), 0d);
+    assertThat(Double.isNaN(reader.nextDouble())).isTrue();
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: `assertThat(reader.nextDouble()).isNaN()`

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java [hunk 7]
```diff
@@ -75,32 +71,32 @@
   @Test public void ContainsNonComparableKeyReturnsFalse() {
     LinkedHashTreeMap<String, String> map = new LinkedHashTreeMap<>();
     map.put("a", "android");
-    assertFalse(map.containsKey(new Object()));
+    assertThat(map.containsKey(new Object())).isFalse();
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: `assertThat(map).doesNotContainKey(new Object());`

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java [hunk 8]
```diff
@@ -129,7 +125,7 @@
     map.put("b", "bbq");
     map.clear();
     assertIterationOrder(map.keySet());
-    assertEquals(0, map.size());
+    assertThat(map.size()).isEqualTo(0);
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: `assertThat(map).isEmpty()`

**Comment by JakeWharton** [comment]
Just style things. Not strictly required. LGTM.

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java (hunk 1 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: I made this by refactoring by making my own Assert class, and then hitting inline on everything. IntelliJ extracted this one.

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java (hunk 2 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: Fixed for all.

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java (hunk 3 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: Fixed.

File: moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java (hunk 6 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/JsonReaderTest.java]: Nice. Fixed.

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java (hunk 5 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: Fixed.

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java (hunk 4 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: Fixed.

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java (hunk 7 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: Fied.

File: moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java (hunk 8 above)
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/LinkedHashTreeMapTest.java]: Fixed.

**Pull request merged**